independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Decorators.java

```diff
@@ -80,6 +83,14 @@ static DecoratorInfo createDecorator(ClassInfo decoratorClass, BeanDeployment be
                 throw new DefinitionException("The delegate type " + delegateInjectionPoint.getRequiredType()
                         + " does not implement the decorated type: " + decoratedType);
             }
+            for (DotName builtInBeanName : builtInBeanTypes) {
+                if (builtInBeanName.equals(decoratedType.name())) {
+                    LOGGER.warn("Decorating built-in bean types is not supported! Decorator " + decoratorClass +
```

Comment:
Hm, I think that we should fail the build instead of a warning...

Reply:
As per my initial comment, this is perfectly fine with me. I'll adjust the PR :)